templates/Coalesce.Vue.Template/content/.github/workflows/build-test-and-deploy.yml

```diff
@@ -0,0 +1,103 @@
+name: Build, Test, and Deploy Coalesce.Starter.Vue
+
+# https://docs.github.com/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+# CONFIGURATION
+# For help, go to https://github.com/Azure/Actions
+#
+# 1. Set up the following secrets in your repository:
+#   AZURE_WEBAPP_PUBLISH_PROFILE
+#
+# 2. Set up the production Environment in the GitHub Repository Settings.
+#   Environment Protection Rules are recommended to prevent accidental deployments.
+#   https://docs.github.com/actions/managing-workflow-runs-and-deployments/managing-deployments/managing-environments-for-deployment#deployment-protection-rules
+#
+# 3. Change these variables for your configuration:
+env:
+  AZURE_WEBAPP_NAME: Coalesce.Starter.Vue # set this to your app service name
```

Comment:
Might put the placeholder value as something that could plausibly be an appservice name (replace dashes with dots). It'll be wrong, but it'll help hint to a confused user that we're looking for the app service name, not the csproj name.

Reply:
Update to `MY_WEBAPP_NAME`

Reply:
Sorry, I said that backwards. I meant replace dots with dashes

Reply:
underscores aren't valid in app service names 
![image](https://github.com/user-attachments/assets/9d31e42b-30a2-4e2a-9bb7-36a3f44e10e3)

Reply:
I figured 😄

Reply:
Offline convo - keeping as an invalid name to match GH docs: https://docs.github.com/en/actions/use-cases-and-examples/deploying/deploying-net-to-azure-app-service#creating-the-workflow

Correct naming guidelines for reference: https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules?WT.mc_id=8B97120A00B57354#microsoftweb